internal/crypto/cert.go

```diff
@@ -59,3 +68,95 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// MergeCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func MergeCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	// get cert bundle path
+	if pluginConfig == nil {
+		return originalCerts, nil
+	}
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	if !ok {
+		return originalCerts, nil
+	}
+
+	// validate certificate bundle path
+	certBundlePath = strings.Trim(certBundlePath, " ")
```

Comment:
This might not be true because a unix file is allowed to be end with a space. For example, in `unix`:
`filepath.Base(" /foo/bar/baz.go ")` returns `baz.go ` with a filename of length 7. And `os.ReadFile('baz.go ')` would be fine.
Since we are already checking cross-platform compatible below, we are explicitly rejecting this kind of naming. Thus, we could just remove the `strings.Trim`.

Reply:
Updated.